lib/post/widgets/post_view.dart

```diff
@@ -24,6 +24,8 @@ class PostSubview extends StatelessWidget {
     final theme = Theme.of(context);
 
     final bool isUserLoggedIn = context.read<AuthBloc>().state.isLoggedIn;
+    final bool hideNsfwPreviews = context.read<ThunderBloc>().state.preferences?.getBool('setting_general_show_save_action') ?? true;
+
```

Comment:
same here - the name of the setting should be `setting_general_hide_nsfw_previews`